saist/main.py

```diff
@@ -42,9 +44,9 @@ async def analyze_single_file(scm: Scm, adapter: BaseLlmAdapter, filename, patch
     """
     system_prompt = prompts.DETECT
```

Comment:
It's the system_prompt which needs the tailored rules applying, not the user prompt which contains the file contents